xnat_tools/bids_utils.py

```diff
@@ -469,23 +469,24 @@ def validate_frame_counts(scans: list, bids_session_dir: str) -> None:
                     key=extract_slice_number,
                 )
 
-            # Compare frame counts of first and last acquisition. Remove the last if unequal.
+            # Compare frame counts of first and all other DICOMs. Remove other DICOMs if unequal.
+            # Should generally only be the last DICOM, unless data is multiecho and/or mag/phase
             if dicom_files:
                 first_frame_count = read_dicom_frame_count(
                     os.path.join(bids_scan_dir, dicom_files[0])
                 )
-                last_frame_count = read_dicom_frame_count(
-                    os.path.join(bids_scan_dir, dicom_files[-1])
-                )
 
-                if last_frame_count != first_frame_count:
-                    last_file_path = os.path.join(bids_scan_dir, dicom_files[-1])
+                for dicomfile in dicom_files[1:]:
```

Comment:
Looks great!  Thanks for working with Apoorva to figure out the issue. Can we gather the protocol (mag/phase or multecho) from the series_desc?  That way we can compare all dicoms in only those specific cases.